.github/workflows/build.yml

```diff
@@ -20,43 +30,38 @@ jobs:
+      - name: Build binary
         run: |
-          sudo apt-get update && sudo apt-get install -y osslsigncode
           cp LICENSE "$RUNNER_TEMP/LICENSE"
           echo -e "\n---\n" >> "$RUNNER_TEMP/LICENSE"
           curl -L "https://go.dev/LICENSE?m=text" >> "$RUNNER_TEMP/LICENSE"
           VERSION="$(git describe --tags)"
-          function build_age() {
-            DIR="$(mktemp -d)"
-            mkdir "$DIR/age"
-            cp "$RUNNER_TEMP/LICENSE" "$DIR/age"
-            go build -o "$DIR/age" -ldflags "-X main.Version=$VERSION" -trimpath ./cmd/...
-            if [ "$GOOS" == "windows" ]; then
-              if [ -n "${{ secrets.SIGN_PASS }}" ]; then
-                for exe in "$DIR"/age/*.exe; do
-                  /usr/bin/osslsigncode sign -t "http://timestamp.comodoca.com" \
-                    -certs .github/workflows/certs/uitacllc.crt \
-                    -key .github/workflows/certs/uitacllc.key \
-                    -pass "${{ secrets.SIGN_PASS }}" \
-                    -n age -in "$exe" -out "$exe.signed"
-                  mv "$exe.signed" "$exe"
-                done
-              fi
-              ( cd "$DIR"; zip age.zip -r age )
-              mv "$DIR/age.zip" "age-$VERSION-$GOOS-$GOARCH.zip"
-            else
-              tar -cvzf "age-$VERSION-$GOOS-$GOARCH.tar.gz" -C "$DIR" age
+          DIR="$(mktemp -d)"
+          mkdir "$DIR/age"
+          cp "$RUNNER_TEMP/LICENSE" "$DIR/age"
+          go build -o "$DIR/age" -ldflags "-X main.Version=$VERSION" -trimpath ./cmd/...
+          if [ "$GOOS" == "windows" ]; then
+            sudo apt-get update && sudo apt-get install -y osslsigncode
+            if [ -n "${{ secrets.SIGN_PASS }}" ]; then
+              for exe in "$DIR"/age/*.exe; do
+                /usr/bin/osslsigncode sign -t "http://timestamp.comodoca.com" \
+                  -certs .github/workflows/certs/uitacllc.crt \
+                  -key .github/workflows/certs/uitacllc.key \
+                  -pass "${{ secrets.SIGN_PASS }}" \
+                  -n age -in "$exe" -out "$exe.signed"
+                mv "$exe.signed" "$exe"
+              done
             fi
-          }
-          export CGO_ENABLED=0
-          GOOS=linux GOARCH=amd64 build_age
-          GOOS=linux GOARCH=arm GOARM=6 build_age
-          GOOS=linux GOARCH=arm64 build_age
-          GOOS=darwin GOARCH=amd64 build_age
-          GOOS=darwin GOARCH=arm64 build_age
-          GOOS=windows GOARCH=amd64 build_age
-          GOOS=freebsd GOARCH=amd64 build_age
+            ( cd "$DIR"; zip age.zip -r age )
+            mv "$DIR/age.zip" "age-$VERSION-$GOOS-$GOARCH.zip"
+          else
+            tar -cvzf "age-$VERSION-$GOOS-$GOARCH.tar.gz" -C "$DIR" age
+          fi
+        env:
+          CGO_ENABLED: 0
+          GOARCH: ${{ matrix.arch }}
+          GOARM: ${{ matrix.arm }}
+          GOOS: ${{ matrix.os }}
       - name: Upload workflow artifacts
         uses: actions/upload-artifact@v2
```

Comment:
So does `actions/upload-artifact@v2` combine multiple uploads?

Reply:
Yes, all the artifacts will be appanded under the same `name`

Reply:
#### Minimal reproducible example

```yaml
on: workflow_dispatch
jobs:
  create:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        file: [1, 2]
    steps:
    - run: date > file-${{ matrix.file }}
    - uses: actions/upload-artifact@v2
      with:
        name: files
        path: file-*
  read:
    needs: create
    runs-on: ubuntu-latest
    steps:
    - uses: actions/download-artifact@v2
      with:
        name: files
    - run: ls

```

Reply:
(Side note: another nice preoperty of artifacts is that you can also overwrite them)